src/MicroPlate.php

```diff
@@ -27,18 +34,133 @@ class MicroPlate
      */
     public function __construct(CoordinateSystem $coordinateSystem)
     {
-        $this->wells = new Collection([]);
         $this->coordinateSystem = $coordinateSystem;
+
+        $this->clearWells();
     }
 
     /**
-     * TODO how to deal with duplicates?
-     *
      * @param Coordinate<TCoordinateSystem> $coordinate
      * @param TWell $content
      */
     public function addWell(Coordinate $coordinate, $content): void
     {
-        $this->wells->add([$coordinate, $content]);
+        $this->assertWellIsEmpty($coordinate, $content);
+        $this->forceAddWell($coordinate, $content);
+    }
+
+    /**
+     * @param TWell $content
+     */
+    public function addToNextFreeWell($content, FlowDirection $flowDirection): void
+    {
+        $coordinate = $this->nextFreeWellCoordinate($flowDirection);
+        $this->addWell($coordinate, $content);
+    }
+
+    public function clearWells(): void
+    {
+        $this->wells = new Collection([]);
+        foreach ($this->coordinateSystem->columns() as $column) {
+            foreach ($this->coordinateSystem->rows() as $row) {
+                $this->wells->put($row . $column, self::EMPTY_WELL);
+            }
+        }
+    }
+
+    /**
+     * @param Coordinate<TCoordinateSystem> $coordinate
+     */
+    public function wellIsEmpty(Coordinate $coordinate): bool
+    {
+        return self::EMPTY_WELL === $this->wells[$coordinate->toString()];
```

Comment:
```suggestion
        return self::EMPTY_WELL === $this->well($coordinate);
```

Coordinate to index conversion is duplicated many times in this class. Is it part of the API contract?